services/MerchStoreService.ts

```diff
@@ -457,4 +461,53 @@ export default class MerchStoreService {
     }
     return requestedQuantitiesByMerchItem;
   }
+
+  public async getPastPickupEvents(canSeeOrders = false): Promise<PublicOrderPickupEvent[]> {
+    const pickupEvents = await this.transactions.readOnly(async (txn) => Repositories
+      .merchOrderPickupEvent(txn)
+      .getPastPickupEvents());
+    return pickupEvents.map((e) => e.getPublicOrderPickupEvent(canSeeOrders));
+  }
+
+  public async getFuturePickupEvents(canSeeOrders = false): Promise<PublicOrderPickupEvent[]> {
+    const pickupEvents = await this.transactions.readOnly(async (txn) => Repositories
+      .merchOrderPickupEvent(txn)
+      .getFuturePickupEvents());
+    return pickupEvents.map((e) => e.getPublicOrderPickupEvent(canSeeOrders));
+  }
+
+  public async createPickupEvent(pickupEvent: OrderPickupEvent): Promise<PublicOrderPickupEvent> {
+    return this.transactions.readWrite(async (txn) => {
+      if (pickupEvent.start >= pickupEvent.end) {
+        throw new UserError('Order pickup event start time must come before the end time');
+      }
+      const createdPickupEvent = await Repositories.merchOrderPickupEvent(txn)
+        .upsertPickupEvent(OrderPickupEventModel.create(pickupEvent));
+      return createdPickupEvent.getPublicOrderPickupEvent();
+    });
+  }
+
+  public async editPickupEvent(uuid: Uuid, changes: OrderPickupEventEdit): Promise<PublicOrderPickupEvent> {
+    return this.transactions.readWrite(async (txn) => {
+      const orderPickupEventRepository = Repositories.merchOrderPickupEvent(txn);
+      const pickupEvent = await orderPickupEventRepository.findByUuid(uuid);
+      const updatedPickupEvent = OrderPickupEventModel.merge(pickupEvent, changes);
+      if (updatedPickupEvent.start >= updatedPickupEvent.end) {
+        throw new UserError('Order pickup event start time must come before the end time');
+      }
+      const upsertedPickupEvent = await orderPickupEventRepository.upsertPickupEvent(updatedPickupEvent);
+      return upsertedPickupEvent.getPublicOrderPickupEvent();
+    });
+  }
+
+  public async deletePickupEvent(uuid: Uuid): Promise<void> {
```

Comment:
How would deleting a pickup event work? Email the attendees, delete the event from their orders, and then delete the event? Should there be a way to do this?

Reply:
Deleting a pickup event is only allowed for events that don't have orders attached to it, so it'd only be used if an admin creates a 'faulty' event and wants to delete it before anyone uses it.

Reply:
What if an admin has to cancel an event for some reason e.g. heavy rain?

Reply:
@steets250 any thoughts on this behavior?

Reply:
Hmm, I think Sumeet's use case is a good idea in the event that we need to cancel a pick up event. Would we want an order status type that reflects the state of an order where it's pickup event (e.g. `PICKUP_CANCELLED`) to mark an order as needing to be rescheduled for pickup, or should frontend just check if the pickup event field is empty and allow the user to set a new one?

Reply:
`PICKUP_CANCELED` sounds like a good idea to me, so that we can differentiate that case and handle it correctly instead of auto-canceling and -refunding that order at the end of the quarter. Maybe some grace period of a month or end of quarter (whichever comes later) to reschedule?

Reply:
I'm going to omit this `deletePickupEvent` change from this PR and implement it in the next PR (which is for handling order states like `MISSED, CANCELLED, etc.` since it seems to be more relevant to that. For now, I'll delete the `DELETE /order/pickup` endpoint so that deletion can't be used

Reply:
tracking in #204